### PR TITLE
Update mongodb crud operations

### PR DESCRIPTION
**Pull request by timotheekelly**
Adding new code to correspond with the updated article https://www.baeldung.com/java-mongodb

### CODE REVIEW

File: persistence-modules/java-mongodb/pom.xml [hunk 1]
```diff
@@ -45,9 +55,43 @@
...
+                            <exclusion>
+                                <groupId>junit</groupId>
+                                <artifactId>junit</artifactId>
+                            </exclusion>
+                        </exclusions>
+                    </dependency>
+                </dependencies>
+            </plugin>
+        </plugins>
+    </build>
```
Comment (theangrydev) [inline persistence-modules/java-mongodb/pom.xml:95]: Remove this, we have defaults in the parent pom.xml I think

File: persistence-modules/java-mongodb/src/main/java/com/baeldung/MongodbExample.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.Projections;
+import com.mongodb.client.model.Sorts;
+import com.mongodb.client.model.Updates;
+
+public class MongodbExample {
```
Comment (theangrydev) [inline persistence-modules/java-mongodb/src/main/java/com/baeldung/MongodbExample.java:25]: This class doesn't seem to be used in any tests? All code included in the PR has to have unit tests - can we remove this and rely on the MongodbCrudOperationsLiveTest?

File: spring-security-modules/spring-security-oauth2-bff/backend/bff/.pmdruleset.xml [hunk 3]
```diff
@@ -1,9 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (theangrydev) [inline spring-security-modules/spring-security-oauth2-bff/backend/bff/.pmdruleset.xml:1]: Undo unrelated changes

File: spring-security-modules/spring-security-oauth2-bff/backend/resource-server/.pmdruleset.xml [hunk 4]
```diff
@@ -1,9 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (theangrydev) [inline spring-security-modules/spring-security-oauth2-bff/backend/resource-server/.pmdruleset.xml:1]: Undo unrelated changes

File: spring-security-modules/spring-security-oauth2-bff/backend/reverse-proxy/.pmdruleset.xml [hunk 5]
```diff
@@ -1,9 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (theangrydev) [inline spring-security-modules/spring-security-oauth2-bff/backend/reverse-proxy/.pmdruleset.xml:1]: Undo unrelated changes

**Review by theangrydev — CHANGES_REQUESTED** [review]

File: persistence-modules/java-mongodb/src/main/java/com/baeldung/MongodbExample.java (hunk 2 above)
Comment (timotheekelly) [reply persistence-modules/java-mongodb/src/main/java/com/baeldung/MongodbExample.java:25]: Absolutely, resolving now

File: persistence-modules/java-mongodb/pom.xml (hunk 1 above)
Comment (timotheekelly) [reply persistence-modules/java-mongodb/pom.xml:95]: Removed

File: spring-security-modules/spring-security-oauth2-bff/backend/reverse-proxy/.pmdruleset.xml (hunk 5 above)
Comment (timotheekelly) [reply spring-security-modules/spring-security-oauth2-bff/backend/reverse-proxy/.pmdruleset.xml]: Resolved

File: spring-security-modules/spring-security-oauth2-bff/backend/resource-server/.pmdruleset.xml (hunk 4 above)
Comment (timotheekelly) [reply spring-security-modules/spring-security-oauth2-bff/backend/resource-server/.pmdruleset.xml]: Resolved

File: spring-security-modules/spring-security-oauth2-bff/backend/bff/.pmdruleset.xml (hunk 3 above)
Comment (timotheekelly) [reply spring-security-modules/spring-security-oauth2-bff/backend/bff/.pmdruleset.xml]: Resolved

**Comment by theangrydev** [comment]
retest this please

File: persistence-modules/java-mongodb/src/test/java/com/baeldung/MongodbCrudOperationsLiveTest.java [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+                assertNotNull(doc.getString("title"));
+                assertNotNull(doc.getInteger("year"));
+                assertNull(doc.get("_id"));
+            }
+        } finally {
+            cursor.close();
+        }
+        if (titles.size() > 1) {
+            for (int i = 0; i < titles.size() - 1; i++) {
+                assertTrue(titles.get(i)
```
Comment (theangrydev) [inline persistence-modules/java-mongodb/src/test/java/com/baeldung/MongodbCrudOperationsLiveTest.java:133]: I just noticed we have quite a few of these complex assertTrue assertions

We have a guideline to switch to using AssertJ if the assertions start to get complex like this - can we refactor the assertions to use AssertJ throughout please?

**Review by theangrydev — CHANGES_REQUESTED** [review]

**Comment by theangrydev** [comment]
retest this please

File: persistence-modules/java-mongodb/src/test/java/com/baeldung/MongodbCrudOperationsLiveTest.java (hunk 6 above)
Comment (theangrydev) [reply persistence-modules/java-mongodb/src/test/java/com/baeldung/MongodbCrudOperationsLiveTest.java:133]: @timotheekelly ping here

Comment (timotheekelly) [reply persistence-modules/java-mongodb/src/test/java/com/baeldung/MongodbCrudOperationsLiveTest.java:133]: Yes I can refactor this now, thank you

Comment (timotheekelly) [reply persistence-modules/java-mongodb/src/test/java/com/baeldung/MongodbCrudOperationsLiveTest.java]: I've refactored to use the AssertJ. I'm somewhat unfamiliar, but I believe these tests follow best practice

**Comment by timotheekelly** [comment]
retest this please

**Comment by theangrydev** [comment]
retest this please

**Review by theangrydev — APPROVED** [review]
